compute_sdk/tests/unit/test_client.py

```diff
@@ -331,6 +342,61 @@ def funk():
     assert func_data.metadata is None
 
 
+def test_register_function_no_function(gcc):
+    gcc.web_client = mock.MagicMock()
+
+    with pytest.raises(ValueError) as pyt_exc:
+        gcc.register_function(None)
+
+    assert "either" in str(pyt_exc).lower()
+    assert "'function'" in str(pyt_exc).lower()
+    assert "'function_name'" in str(pyt_exc).lower()
+    assert "'function_code'" in str(pyt_exc).lower()
+
+
+def test_function_registration_data_function():
+    frd = FunctionRegistrationData(function=funk)
+
+    assert frd.function_name == "funk"
+    assert frd.function_code == _get_packed_code(funk)
+
+
+def test_function_registration_data_function_name_and_code():
+    frd = FunctionRegistrationData(
+        function=None, function_name="foo", function_code="bar"
+    )
```

Comment:
Not a big deal, `foo` and `bar` are well-known, but in the [spirit of #dieselgate](https://en.wikipedia.org/wiki/Volkswagen_emissions_scandal), I often like to show "there's nothing up my sleeve" in tests by using something akin to `randomstring()` rather than static strings.